### PR TITLE
Fix/warn file too big

### PR DESCRIPTION
**Pull request by petersg83**
No modification where made for the providers `aws` and `rackspace` as they accept files up to 5 Go and the node buffer limit is 2 Go. So I couldn't test files bigger than 5 Go + the error would throw before the provider says it's too large.
It will be needed once we have change the buffers for streams.

@soupette @HichamELBSI I tag you because I updated the front part: the linter was not happy. Can you check everything's fine please?

### CODE REVIEW

File: packages/strapi/lib/middlewares/parser/index.js [hunk 1]
```diff
@@ -37,14 +37,32 @@ module.exports = strapi => {
...
+        }
+
+        try {
+          const res = await body({
+            patchKoa: true,
+            ...omit(strapi.config.middleware.settings.parser, 'queryStringParser'),
+          })(ctx, next);
+          return res;
+        } catch (e) {
+          if (e.message.includes('maxFileSize exceeded')) {
```
Comment (alexandrebodin) [inline packages/strapi/lib/middlewares/parser/index.js:54]: formidable doesn't throw already a 413 ?

**Review by alexandrebodin — CHANGES_REQUESTED** [review]
I don't see any changes to the aws and rackspace providers to handle this ? any reason ?

File: packages/strapi/lib/middlewares/parser/index.js (hunk 1 above)
Comment (petersg83) [reply packages/strapi/lib/middlewares/parser/index.js:54]: No it was just throwing an error, resulting to an error 500

File: packages/strapi-plugin-upload/config/functions/bootstrap.js [hunk 2]
```diff
@@ -31,11 +28,37 @@ module.exports = async () => {
   registerPermissionActions();
 };
 
+const wrapFunctionForErrors = fn => async (...args) => {
+  try {
+    return await fn(...args);
+  } catch (err) {
```
Comment (alexandrebodin) [inline packages/strapi-plugin-upload/config/functions/bootstrap.js:34]: In term of design here is what I think would be nice to have:


```js
try {

} catch (err) {
 if (errors.is(err, errors.ENTITY_TOO_LARGE)) {
  throw entityTooLarge
 } else {
 throw badImplementation
}
}

```


using this allow to encapsultabe the checking logic (e.g err.type)

File: packages/strapi-plugin-upload/constants.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+const errorTypes = {
```
Comment (alexandrebodin) [inline packages/strapi-plugin-upload/constants.js:1]: I would move that to the errors file and only expose the errors interface

**Review by alexandrebodin — CHANGES_REQUESTED** [review]

**Comment by soupette** [comment]
@petersg83 seems like the linter is happy now.

File: packages/strapi-plugin-upload/config/functions/bootstrap.js [hunk 4]
```diff
@@ -31,11 +28,35 @@ module.exports = async () => {
...
+    if (isError(err, errors.entityTooLarge)) {
```
Comment (alexandrebodin) [inline packages/strapi-plugin-upload/config/functions/bootstrap.js:35]: I think you should move that conversion logic in the errors module or at least outside of the wrapFunction util

**Review by alexandrebodin — CHANGES_REQUESTED** [review]
Last comment :)

**Review by alexandrebodin — APPROVED** [review]

**Pull request merged**